sdk/dotnet/Pulumi/Core/OutputUtilities.cs

```diff
@@ -23,6 +23,15 @@ public static async Task<bool> GetIsKnownAsync<T>(Output<T> output)
             return data.IsKnown;
         }
 
+        /// <summary>
```

Comment:
This facility is intended for use in test and debugging scenarios. In normal Pulumi programs, please consider using `.Apply` instead to chain `Output<T>` transformations without unpacking the underlying `T`. Doing so preserves metadata such as resource dependencies that is used by Pulumi engine to operate correctly. Using `await o.GetValueAsync()` directly opens up a possibility to introduce issues with lost metadata.

Reply:
I agree this should be avoided in general. Should I change the code comment to read as above?

Reply:
I can do that for you :) Let's merge this up and I can change the comment in the follow up PR.